### ISSUE TITLE
PaginatorHelper.last() method displays "warning" in the screen.

### ISSUE THREAD

**Issue by kondo-abridge** [cakephp/cakephp#11604]
This is a (multiple allowed):

* [x] bug

* CakePHP Version: Perhaps ALL (currently not fixed).
* Platform and Target: CentOS

### What you did
Use "Last" button in my view (with label specified)

### What happened
Display warning: 
"A non-numeric value encountered [CORE/Cake/View/Helper/PaginatorHelper.php, line ...

### What you expected to happen
> public function last($last = 'last >>', array $options = [])

Function last's argument $last may be "label string". not only "int string"
But in this function, 

>  $lower = (int)$params['pageCount'] - (int)$last + 1;

This should be fixed. Like this
```php
 if (is_int($last)) {
  $lower = (int)$params['pageCount'] - (int)$last + 1; 
} else {
  $lower = (int)$params['pageCount']  + 1;
}
```
That's all.

**Comment by dereuromark** [comment]
Seems like there is a test missing for this strange mix of type allowance here.

**Comment by dereuromark** [comment]
Are you able to provide a PR here?

PS: `if (is_numeric($last)) {` must be used instead to be BC - as you can pass `'1'` which still should work.

**Comment by markstory** [comment]
How can someone reproduce the warning you're seeing? I'm not entirely clear on it.

**Comment by saeideng** [comment]
@dereuromark 
`if (is_numeric($last))` :-1: 

right now '1' not acceptable

**Comment by dereuromark** [comment]
I disagree. Check the results of the methods. you will see that i am right.

**Comment by saeideng** [comment]
I have 3 page
```php
$this->Paginator->last('2');
//return a link with label '2' and link to latest page (3)
```
```php
$this->Paginator->last(2);
//return 2 link with label '2' and '3' link to (2) and (3)
```

**Comment by saeideng** [comment]
but why?
because we used `is_int` here
https://github.com/cakephp/cakephp/blob/28400cb7baf74a8dc9abf2e13fe76bacccf54ec0/src/View/Helper/PaginatorHelper.php#L1084

**Comment by dereuromark** [comment]
is_int() with string integers (if a good idea or not) will not work as before, thus it wouldnt be BC.

**Comment by saeideng** [comment]
```php
 if (is_int($last)) {
  $lower = (int)$params['pageCount'] - (int)$last + 1; 
} else {
  $lower = (int)$params['pageCount']  + 1;
}
```
and
```php
 if (is_numeric($last)) {
  $lower = (int)$params['pageCount'] - (int)$last + 1; 
} else {
  $lower = (int)$params['pageCount']  + 1;
}
```
both above code , will not change behavior of 
```php
$this->Paginator->last('2');
//and
$this->Paginator->last('2');
```
so , you can use both `is_numeric` and `is_int` here

also 
```php
(int)'string' // returns 0 
```
and I'm not sure  that this issue is true 
I say, we no need to change the core

**Comment by dereuromark** [comment]
Is it possible that the PHP behavior changed in a more recent version? Also in regards to strict type?
I think we need to know more details about the environment this was run. Because that will most likely matter for reproducing this.

**Comment by saeideng** [comment]
I a bit tested 
` (int)(string)`
In all php version `0`
without error/warning

**Comment by dereuromark** [comment]
I meant the is_int() method. I am 100% certain that it is quite strict, at least in older PHP versions.

**Comment by markstory** [comment]
`is_int()` hasn't changed its behavior. There is [an existing test covering the behavior](https://github.com/cakephp/cakephp/blob/28400cb7baf74a8dc9abf2e13fe76bacccf54ec0/tests/TestCase/View/Helper/PaginatorHelperTest.php#L2303-L2311) described here. Making string integers work the same as actual integers would break existing behavior.

**Comment by ADmad** [comment]
I don't think anything needs to be done here. Allowing both integer and strings is wacky enough, let's not complicate it anymore.

 In 4.x we should consider having two separate methods for int and string params.

**Comment by markstory** [comment]
> In 4.x we should consider having two separate methods for int and string params.

Yes. We can add more logically behaving methods in the future.

**Comment by chinpei215** [comment]
@kondo-abridge Please tell us what version of CakePHP you are using. Reading the error message I think you are using 2.x, as the file path contains `Cake`. Then I think the issue has been already fixed in #9592.

**Comment by kondo-abridge** [comment]
Yes, I am using 2.7.6. Thank you.
